cli/cmd/migrate_webhooks.go

```diff
@@ -0,0 +1,464 @@
+			if err != nil {
+				return fmt.Errorf("cannot retrieve webhook-config for project %s: %s", project.ProjectName, err.Error())
+			}
+			if err := migrateWebhooks(webhooks, params, api); err != nil {
+				return fmt.Errorf("cannot migrate webhook-config for project %s: %s", project.ProjectName, err.Error())
+			}
+		}
+	}
+
+	return nil
+}
+
+func getProjectWebhooks(project *models.Project, api *api.APISet) ([]*webhookResource, error) {
+	var webhooks []*webhookResource
+
+	// getting project resources
+	projectWebhookResource, err := api.ResourcesV1().GetProjectResource(project.ProjectName, webhookURI)
+	if err != nil && errors.Is(err, ErrResourceNotFound) {
+		return nil, fmt.Errorf("cannot retrieve webhook resource on project level for project %s: %s", project.ProjectName, err.Error())
+	}
+	if projectWebhookResource != nil {
+		tmpWebhookResource := webhookResource{
+			Project:         &project.ProjectName,
+			WebhookResource: projectWebhookResource,
+		}
+		webhooks = append(webhooks, &tmpWebhookResource)
+	}
+
+	// getting stage resources
+	for _, stage := range project.Stages {
+		stageWebhookResource, err := api.ResourcesV1().GetStageResource(project.ProjectName, stage.StageName, webhookURI)
+		if err != nil && errors.Is(err, ErrResourceNotFound) {
+			return nil, fmt.Errorf("cannot retrieve webhook resource on stage level for project %s: %s", project.ProjectName, err.Error())
+		}
+		if stageWebhookResource != nil {
+			tmpWebhookResource := webhookResource{
+				Project:         &project.ProjectName,
+				Stage:           &stage.StageName,
+				WebhookResource: stageWebhookResource,
+			}
+			webhooks = append(webhooks, &tmpWebhookResource)
+		}
+	}
+
+	// getting service resources
+	for _, stage := range project.Stages {
+		for _, service := range stage.Services {
+			serviceWebhookResource, err := api.ResourcesV1().GetServiceResource(project.ProjectName, stage.StageName, service.ServiceName, webhookURI)
+			if err != nil && errors.Is(err, ErrResourceNotFound) {
+				return nil, fmt.Errorf("cannot retrieve webhook resource on service level for project %s: %s", project.ProjectName, err.Error())
+			}
+			if serviceWebhookResource != nil {
+				tmpWebhookResource := webhookResource{
+					Project:         &project.ProjectName,
+					Stage:           &stage.StageName,
+					Service:         &service.ServiceName,
+					WebhookResource: serviceWebhookResource,
+				}
+				webhooks = append(webhooks, &tmpWebhookResource)
+			}
+		}
+	}
+
+	return webhooks, nil
+}
+
+func migrateWebhooks(webhooks []*webhookResource, params *migrateWebhooksCmdParams, api *api.APISet) error {
+	for _, w := range webhooks {
+		webhook := resourceToWebhook(w.WebhookResource)
+		if webhook == nil {
+			return fmt.Errorf("cannot decode webhook resource")
+		}
+		// migrate only webhooks in v1alpha1 version
+		if webhook.ApiVersion == betaApiVersion {
+			continue
+		}
+		migratedWebhook, err := migrateAlphaWebhook(webhook)
+		if err != nil {
+			return err
```

Comment:
here I suggest to log the error, but continue afterwards - otherwise one corrupt webhook resource will prevent all others from being migrated